chalk-engine/src/logic.rs

```diff
@@ -112,7 +112,7 @@ impl<C: Context> Forest<C> {
         mut test: impl FnMut(&C::InferenceNormalizedSubst) -> bool,
     ) -> bool {
         if let Some(answer) = self.tables[table].answer(answer) {
-            info!("answer cached = {:?}", answer);
+            info!(?answer, "answer cached");
```

Comment:
Woah! what is happening here. I guess this is a different for how tracing works?

Reply:
Yup, in tracing the logging macros generate `Event`s which may contain a number of fields. The `?` sigil is a shorthand for creating a new field using the `Debug` implementation of `answer`.